linq4j/src/main/java/org/apache/calcite/linq4j/tree/Expressions.java

```diff
@@ -622,9 +622,49 @@ public static UnaryExpression convert_(Expression expression, Type type,
    * operation that throws an exception if the target type is
    * overflowed.
    */
-  public static UnaryExpression convertChecked(Expression expression,
+  public static Expression convertChecked(Expression expression,
       Type type) {
-    throw Extensions.todo();
+    if (type == Byte.class
```

Comment:
It's funny, I implemented this myself in a separate PR: https://github.com/apache/calcite/pull/3589
There I am relying on an existing implementation in Primitive.numberValue.

Reply:
Nice. Your approach seems to cover more scenarios.

I will keep `convertChecked` as-is for now. The inlined validations should be sufficient within the current PR's scope, but feel free to replace it once #3589 has been merged.